src/environment.js

```diff
@@ -120,7 +119,11 @@ var Environment = Obj.extend({
             throw new Error('template names must be a string: ' + name);
         }
 
-        var tmpl = this.cache[name];
+        for (var i = 0; i < this.loaders.length; i++) {
+            var _name = parentName? this.loaders[i].resolve(parentName, name) : name;
+            tmpl = this.loaders[i].cache[_name];
```

Comment:
This is related to https://github.com/mozilla/nunjucks/issues/378#issuecomment-75380107